x-pack/test/apm_api_integration/common/config.ts

```diff
@@ -43,15 +48,45 @@ async function getApmApiClient({
 
 export type CreateTestConfig = ReturnType<typeof createTestConfig>;
 
-export function createTestConfig(config: ApmFtrConfig) {
+type ApmApiClientKey =
+  | 'noAccessUser'
+  | 'readUser'
+  | 'writeUser'
+  | 'annotationWriterUser'
+  | 'noMlAccessUser'
+  | 'manageOwnAgentKeysUser'
+  | 'createAndAllAgentKeysUser'
+  | 'monitorClusterAndIndicesUser';
+
+export interface CreateTest {
+  testFiles: string[];
+  servers: any;
+  servicesRequiredForTestAnalysis: string[];
+  services: InheritedServices & {
+    apmFtrConfig: () => ApmFtrConfig;
+    registry: ({ getService }: FtrProviderContext) => ReturnType<typeof RegistryProvider>;
+    synthtraceEsClient: (context: InheritedFtrProviderContext) => Promise<ApmSynthtraceEsClient>;
+    apmApiClient: (
+      context: InheritedFtrProviderContext
+    ) => Record<ApmApiClientKey, Awaited<ReturnType<typeof getApmApiClient>>>;
+    ml: ({ getService }: FtrProviderContext) => ReturnType<typeof MachineLearningAPIProvider>;
+  };
+  junit: { reportName: string };
+  esTestCluster: any;
+  kbnTestServer: any;
+}
+
+export function createTestConfig(
+  config: ApmFtrConfig
+): ({ readConfigFile }: FtrConfigProviderContext) => Promise<CreateTest> {
```

Comment:
what's the point of this explicit return type?

Reply:
Build was [failing](https://buildkite.com/elastic/kibana-pull-request/builds/84926#01843a72-1305-440b-ac34-2a3aa9b99e62) here.
`The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.`

Reply:
tbh, I haven't been able to find the culprit that triggered the failure there. The changes related to this `package` includes only adding the agent explorer spec

Reply:
Can you revert here and see if it happens again? Otherwise I'm fine with keeping it but I can't explain it.

Reply:
When I revert the change I keep getting 

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/1313018/201620008-0dc7c58b-648b-4e20-8a8e-0a590b0a8e92.png">